src/main/java/app/dbService/DBService.java

```diff
@@ -0,0 +1,66 @@
+package app.dbService;
+
+import app.dbService.dao.BookDAO;
+import app.dbService.dataSets.Book;
+import org.h2.jdbcx.JdbcDataSource;
+
+import javax.annotation.Resource;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class DBService {
+    private final Connection connection;
+
+    public DBService() {
+        this.connection = getPostgresConnection();
+    }
+
+    public static Connection getPostgresConnection() {
+        try {
+            DriverManager.registerDriver((Driver) Class.forName("org.postgresql.Driver").newInstance());
+            String url = "jdbc:postgresql://localhost:5432/books";
+            String name = "postgres";
+            String pass = "5325475lol";
+
+            JdbcDataSource ds = new JdbcDataSource();
+            ds.setURL(url);
+            ds.setUser(name);
+            ds.setPassword(pass);
+
+
+            Connection connection = DriverManager.getConnection(url, name, pass);
+            return connection;
+        } catch (SQLException | InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public void addBook(Book book) throws DBException{
+        try {
+            connection.setAutoCommit(false);
```

Comment:
i read that to improve performance, you need to turn off auto-commit mode for transaction management

Reply:
Thing is: only worry about such performance improvements only in case your server is slow at the moment. Premature optimization only leads to problems. In your case you have one connection, on which you execute your statements. If you have >1 people using your webservice, you wont be able to guarantee the execution flow that you wanted